### PR TITLE
Trigger CI on push or pull_request but not both

### PR DESCRIPTION
**Pull request by andersy005**
- [ ] Addresses https://github.com/pydata/xarray/pull/5139#issuecomment-817362317
- [x] Passes `pre-commit run --all-files`

Here's a snippet of how this looks like when a pull request is created on the same main repository:

Screen Shot 2021-04-12 at 8 10 27 AM

Cc @keewis

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -9,10 +9,22 @@ on:
...
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          concurrent_skipping: "same_content"
+          skip_after_successful_duplicate: "true"
+          do_not_skip: '["pull_request", "workflow_dispatch"]'
```
Comment (andersy005) [inline .github/workflows/ci.yaml:22]: Incidentally, this action can be used to automatically skip CI runs for documentation related PRs (without needing to use the `[skip-ci]` in the commit message). 


```yaml
with:
	paths_ignore: '["**/doc/**"]'
```

Should we enable this?

Comment (keewis) [reply .github/workflows/ci.yaml:22]: > Should we enable this?

:+1:, it's really easy to forget adding that flag

Comment (max-sixty) [reply .github/workflows/ci.yaml]: Great idea!

File: .github/workflows/ci-pre-commit.yml [hunk 2]
```diff
@@ -5,10 +5,26 @@ on:
...
+
 
 jobs:
+  pre-job:
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
```
[the remark this answers is not in the document]
Comment (andersy005) [reply .github/workflows/ci-pre-commit.yml]: 👍🏽 for using tags. The latest commit includes a fix for this.

**Comment by andersy005** [comment]
Okay, I believe this is ready. I may have missed some corner cases (I'm happy to address these in separate PRs). 



Screen Shot 2022-01-14 at 5 41 47 PM


Screen Shot 2022-01-14 at 5 42 00 PM

**Comment by dcherian** [comment]
Thanks @andersy005

**Pull request merged**